casper/src/main/scala/coop/rchain/casper/dag/BlockDagKeyValueStorage.scala

```diff
@@ -233,6 +241,32 @@ final class BlockDagKeyValueStorage[F[_]: Concurrent: Log] private (
   override def addDeploy(d: Signed[DeployData]): F[Unit] = deployStore.put(d.sig, d)
 
   override def pooledDeploys: F[Map[DeployId, Signed[DeployData]]] = deployStore.toMap
+
+  // Map of deploys being executed and execution results
+  private val execMap    = TrieMap.empty[DeployId, Option[String]]
+  private val expiredMap = TrieMap.empty[DeployId, Unit]
+  def deployStatus(d: DeployId): F[String] =
+    for {
+      finSet <- blockMetadataIndex.finalizedBlockSet
+      r <- BlockDagKeyValueStorage.deployStatus(
+            d,
+            deployStore,
+            deployIndex,
+            invalidBlocksIndex,
+            finSet,
+            execMap.toMap,
+            expiredMap.toMap.keySet
+          )
+    } yield r
+
+  def commitExecutionStarted(
+      d: DeployId
+  ): F[Unit] = Sync[F].delay { execMap.update(d, none[String]) }
+
+  def commitExecutionComplete(
+      d: DeployId,
+      status: String
+  ): F[Unit] = Sync[F].delay { execMap.update(d, status.some) }
```

Comment:
I would prefer to put the `commitExecutionStarted` and `commitExecutionComplete` functions into an trait. This will make the code more consistent and make it easy to put another implementation of `BlockDagStorage`, such as mock.

Reply:
https://github.com/rchain/rchain/pull/3682#discussion_r869619896
I think these are functions just for this particular implementation.